### PR TITLE
[WB-5516] handle missing diff.patch in `wandb restore`

### PR DESCRIPTION
**Pull request by minyoung**
https://wandb.atlassian.net/browse/WB-5516

Description
-----------

When we execute `wandb restore`, the client requests for the `diff.patch` file (amongst other things). In the case of a missing file, the server will return an 'open' file handle, which can be used to upload data to. However, this url will return a `404 Not Found` if we try and read it.

Instead of the client explicitly asking for the files (which results in the open file handle), let's instead ask for the files present.

Testing
-------

Automated testing: `tox -e py38 -- tests/test_cli.py -k restore_no_diff`
Manual testing: `wandb restore --entity $ENTITY $RUN_ID` with a run that has no changes

### CODE REVIEW

**Review by annirudh — APPROVED** [review]
Looks great! I would defer to @raubitsj for any comments he might have, but from what I can tell this is a nice and clean fix. Great job!

**Review by raubitsj — COMMENTED** [review]
@minyoung Was it considered to change the restore code to get a list of files first rather than blindly trying to download diff.patch and wandb-metadata.json like the existing code does?

**Comment by KyleGoyette** [comment]
I suggested this as a fix. My thought process was that it would reduce the number of requests. But checking to see if the diff.patch file exists via a query would also work.

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -576,7 +576,7 @@ def run_config(self, project, run=None, entity=None):
                 bucket(name: $run) {
                     config
                     commit
-                    files(names: ["wandb-metadata.json", "diff.patch"]) {
+                    files {
```
Comment (minyoung) [inline wandb/sdk/internal/internal_api.py:579]: We do support passing in a pattern (which also doesn't create files), but that is resolved using the [LIKE operator](https://github.com/wandb/core/blob/aaffc23943bbd209b13572a95551b6dd6352bdfc/services/gorilla/mysql/metadata.go#L10128), and not a regex. So I don't think that we'd be able to construct a pattern to filter for just the 2 files.

Is getting all the files fine? Or should I add the ability to provide a regex to the server?

**Comment by vanpelt** [comment]
Some runs have 1000’s of files so you would likely need to paginate and it
would be slow.

**Comment by minyoung** [comment]
> Some runs have 1000’s of files so you would likely need to paginate and it would be slow.

Added in pagination. But now I'm wondering if it would be better to make 2 requests using `pattern` for the 2 files? Doing would be more efficient for the case of 1000's of files, but would be less so for projects with less than $page_size of files (but still only 2 requests)

**Comment by minyoung** [comment]
I think that always making 2 requests is a reasonable compromise in terms of performance for large vs small projects. Happy to change it back to paginating through all files if we want to optimize for smaller projects

**Review by raubitsj — APPROVED** [review]
Great.

**Pull request merged**